### PR TITLE
Add lshw partial implementation

### LINKED ISSUE #1717: Partly implement 'lshw' command
Add 'lshw' command which reporting about hardware in the system. Now show only pci, network and usb devices

### PR DESCRIPTION
**Pull request by avinal**
This merge request does the following:-
- Adds  partial `lshw` command implementation
- Closes #1717

### CODE REVIEW

**Comment by anton-bondarev** [comment]
@avinal Thanks. Could you also please move `cpuinfo` and `lsblk` parts to separate procedures. And format a little more carefully.

**Comment by avinal** [comment]
Hello @anton-bondarev Sure, actually I had it added in my development branch, somehow missed in the pull request. I will add it. 

If you insist a more cleaner code I can refactor this code into totally seperate functions for each of the command and then we can just call them in `main`. 

Also I tried keeping the formatting as close to the embox formatting, don't know where I missed. I am using `llvm` format. I will do that again and let you know if I need some help. 

Thanks :)

File: src/cmds/hardware/lshw/lshw.c [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+			usb_dev->usb_iface[0]->iface_desc[0]->b_num_endpoints,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_class,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_subclass,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_protocol,
+			usb_dev->usb_iface[0]->iface_desc[0]->i_interface
+	);
+}
+
+static void show_usb_desc_configuration(struct usb_dev *usb_dev) {
+	struct usb_desc_configuration *config = \
```
Comment (anton-bondarev) [inline src/cmds/hardware/lshw/lshw.c:127]: Why do you use '\' here?

**Comment by anton-bondarev** [comment]
>  I will add it.

Ok.

> If you insist a more cleaner code I can refactor this code into totally seperate functions for each of the command and then we can just call them in `main`.

Yes, I think It will look better and easier to read, Please do it if you could

> Also I tried keeping the formatting as close to the embox formatting, don't know where I missed. I am using `llvm` format. I will do that again and let you know if I need some help.

Thanks. Your last commit is that I meant

File: src/cmds/hardware/lshw/lshw.c (hunk 1 above)
Comment (avinal) [reply src/cmds/hardware/lshw/lshw.c:127]: I didn't get your point. This is the exact code, I didn't change anything. But the backslash can be removed.

**Comment by avinal** [comment]
> Yes, I think It will look better and easier to read, Please do it if you could

On it, I got an unexpected error too I am fixing that too.

File: src/cmds/hardware/lshw/lshw.c (hunk 1 above)
Comment (anton-bondarev) [reply src/cmds/hardware/lshw/lshw.c:127]: Oh, sorry. I didn't understand that it is a cloned code. But in any way, if you use code, it is yours

**Comment by avinal** [comment]
Hello @anton-bondarev 
Sorry for the delay, I have cleaned the code and refactored into seperate functions for each command, and removed unnecessary flag options since lshw is generally run in verbose mode. 

I have also tried to keep the formatting strictly as described in Wiki. Hope this is complete now :)

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Comment by anton-bondarev** [comment]
Hello @avinal Thanks. It's OK for now. I've approved it. Please rebase and squash commits if it's necessary

**Comment by avinal** [comment]
I have rebased and squashed the commits :)

**Comment by anton-bondarev** [comment]
> I have rebased and squashed the commits :)

Thanks. Could you please split on two commits, first lshw command and second inserting it to mods.conf?

**Comment by avinal** [comment]
Done, now I think the version control is really difficult to master than coding itself 😂.

**Comment by anton-bondarev** [comment]
@avinal Thanks.
git workflow is the simplest thing. Embox is a operating system :)

**Pull request merged**